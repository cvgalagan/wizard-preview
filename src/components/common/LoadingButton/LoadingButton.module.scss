@use "../../../styles/common";

.loadingButton {
  position: relative;
  $self: &;
  &__spinnerContainer {
    position: absolute;
    transform: translate(-50%, -50%);
    bottom: 2px;
    left: 50%;
    display: none;
    width: common.$loading-spinner-size;
    height: common.$loading-spinner-size;
  }
  &__spinner {
    width: common.$loading-spinner-size;
    height: common.$loading-spinner-size;
    &_dark {
      color: common.$primary-color;
    }
  }
  &__content {
    visibility: visible;
  }
  &_isLoading {
    #{$self}__content {
      visibility: hidden;
    }
    #{$self}__spinnerContainer {
      display: block;
    }
  }
}