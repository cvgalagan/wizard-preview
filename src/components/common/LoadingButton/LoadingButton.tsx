import React from "react"
import styles from "./LoadingButton.module.scss"
import Button, { ButtonProps } from "../Button/Button"
import Spinner from "../Spinner/Spinner"
import cn from "classnames"

export interface LoadingButtonProps extends ButtonProps {
    loading?: boolean
}

const LoadingButton: React.FC<LoadingButtonProps> = props => {
    const { loading = false, children, className, variant = "primary", ...buttonProps } = props
    return (
        <Button
            {...buttonProps}
            className={cn(styles.loadingButton, loading && styles.loadingButton_isLoading, className)}
            disabled={loading ? true : buttonProps.disabled}
            variant={variant}
        >
            <div className={styles.loadingButton__content}>{children}</div>
            <div className={styles.loadingButton__spinnerContainer}>
                <Spinner
                    className={cn(
                        styles.loadingButton__spinner,
                        variant === "light" || variant === "link" || variant?.startsWith("outline-")
                            ? styles.loadingButton__spinner_dark
                            : undefined
                    )}
                />
            </div>
        </Button>
    )
}

export default LoadingButton
