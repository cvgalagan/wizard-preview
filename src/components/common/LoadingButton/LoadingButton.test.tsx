import React from "react"
import { shallow } from "enzyme"
import LoadingButton from "./LoadingButton"

describe("LoadingButton", () => {
    it("should render in loading state", () => {
        const wrapper = shallow(
            <LoadingButton loading variant="primary">
                Test button
            </LoadingButton>
        )
        expect(wrapper.prop("disabled")).toBeTruthy()
        expect(wrapper).toMatchSnapshot()
    })
})
