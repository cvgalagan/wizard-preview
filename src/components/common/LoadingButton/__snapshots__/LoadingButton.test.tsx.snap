// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton should render in loading state 1`] = `
<Button
  active={false}
  className="loadingButton loadingButton_isLoading"
  disabled={true}
  variant="primary"
>
  <div
    className="loadingButton__content"
  >
    Test button
  </div>
  <div
    className="loadingButton__spinnerContainer"
  >
    <ForwardRef
      className="loadingButton__spinner"
    />
  </div>
</Button>
`;
